pkg/controller/replicaset/replica_set.go

```diff
@@ -198,6 +203,17 @@ func (rsc *ReplicaSetController) Run(workers int, stopCh <-chan struct{}) {
 // getPodReplicaSet returns the replica set managing the given pod.
 // TODO: Surface that we are ignoring multiple replica sets for a single pod.
 func (rsc *ReplicaSetController) getPodReplicaSet(pod *api.Pod) *extensions.ReplicaSet {
+	// look up in the cache, if cached and the cache is valid, just return cached value
+	// Note: if we create a new RS that is also matching a given pod, it will just be ignored
+	// (assuming that the previously matching RS didn't changed), but we sort overlapping RSs
+	// by creation timestamp, so it's no problem. If we change the sort order, we may need
+	// change the code here too.
```

Comment:
Maybe I didn't understand your point, but I think it doesn't help.
In the example above, you have a matching:

pod -> RS2

What happens there is that neither pod nor RS2 changes (only RS1), but this should change the mapping to:

pod -> RS1